e2e_test/e2e_test.go

```diff
@@ -369,6 +370,125 @@ func TestStartStopValidators(t *testing.T) {
 
 }
 
+func runStopNetworkAtL2BlockTest(ctx context.Context, t *testing.T, network test.Network, l2Block *big.Int) {
+	err := network.AwaitBlock(ctx, l2Block.Uint64()-1)
+	require.NoError(t, err)
+
+	shortCtx, cancel := context.WithTimeout(ctx, time.Second*2)
+	defer cancel()
+
+	// fail if any node adds a new block >= the migration block
+	var wg sync.WaitGroup
+	for _, n := range network {
+		wg.Add(1)
+		go func(n *test.Node) {
+			defer wg.Done()
+			err = n.Tracker.AwaitBlock(shortCtx, l2Block.Uint64())
```

Comment:
```suggestion
			err := n.Tracker.AwaitBlock(shortCtx, l2Block.Uint64())
```
There was a data race on this error variable, since it was being accessed concurrently without any synchronisation, redeclaring it here solves the issue.